backend/src/middleware/notifications/notificationsMiddleware.ts

```diff
@@ -120,20 +129,26 @@ const handleContentDataOfPost = async (resolve, root, args, context, resolveInfo
   const idsOfUsers = extractMentionedUsers(args.content)
   const post = await resolve(root, args, context, resolveInfo)
   if (post) {
-    await publishNotifications(
+    const sentEmails: string[] = await publishNotifications(
```

Comment:
This is just a stupid suggestion, but I believe the code is more readable after. Apply only if you agree.
```suggestion
  if (post) {
    const sentEmails: string[] = await publishNotifications(
      context,
      [notifyUsersOfMention('Post', post.id, idsOfUsers, 'mentioned_in_post', context)],
      'emailNotificationsMention',
    )
    
    const sendEmailsFollow =
      await publishNotifications(
        context,
        [notifyFollowingUsers(post.id, groupId, context)],
        'emailNotificationsFollowingUsers',
        sentEmails,
      )
    sentEmails.concat(sendEmailsFollow)
    
    const sentEmailsGroup = 
      await publishNotifications(
        context,
        [notifyGroupMembersOfNewPost(post.id, groupId, context)],
        'emailNotificationsPostInGroup',
        sentEmails,
      )
      sentEmails.concat(sentEmailsGroup) // TODO: this is not even needed, not here, not in your code.
  }
```